### DISCUSSION TITLE
Pin Minor and only update Patches

### DISCUSSION THREAD

**Discussion by carlos-lehmann** [renovatebot/renovate#27484] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.203.3

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### What is your question?

I'm struggling to understand how I can achieve a pin on minor but still update patches. Our use case is to pin Kong to LTS 3.4 docker image. The tag scheme is 3.4.x.x

```
"regexManagers": [
    {
      "datasourceTemplate": "docker",
      "depNameTemplate": "kong/kong-gateway",
      "fileMatch": [
        "^helmfile\\.yaml\\.gotmpl$"
      ],
      "matchStrings": [
        "- image:\\n    tag: (?<currentValue>\\S+)\\n"
      ]
    }
  ],
```

```
  "packageRules": [
    {
      "assignAutomerge": false,
      "automerge": true,
      "commitMessagePrefix": "[release-patch]",
      "gitLabIgnoreApprovals": true,
      "matchPackageNames": [
        "kong/kong-gateway"
      ],
      "matchUpdateTypes": [
        "patch"
      ],
    }
  ],
```
Log Level Debug:
```
         {
           "branchName": "renovate/kong-kong-gateway-3.x",
           "prNo": null,
           "prTitle": "Update kong/kong-gateway Docker tag to v3.6.0.0",
           "result": "no-work",
           "upgrades": [
             {
               "datasource": "docker",
               "depName": "kong/kong-gateway",
               "displayPending": "",
               "fixedVersion": "3.4.3.3",
               "currentVersion": "3.4.3.3",
               "currentValue": "3.4.3.3",
               "newValue": "3.6.0.0",
               "newVersion": "3.6.0.0",
               "packageFile": "helmfile.yaml.gotmpl",
               "updateType": "minor",
               "packageName": "kong/kong-gateway"
             }
           ]
         }
```
it still seems to want to update to 3.6.0.0 which leads me to believe that the version scheme is not recognized? I played around a little bit with regex but I am a bit lost, so any help in finding a solution or tip to get on the right path would be greatly appreciated.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by RahulGautamSingh** [comment, marked as answer]
[`allowedVersions`](https://docs.renovatebot.com/configuration-options/#allowedversions) is one way to achieve this

**Comment by rarkins** [comment]
I'm guessing that you think matchUpdateTypes=patch means "do only patch updates" but that is not the case. packageRules are designed to be like "if X then do Y". matchers are part of X. So your rule means "if updateType is patch then...." not "only do patch updates".

You need to set separateMinorPatch=true for this package and then a second rule which says matchUpdateTypes=minor,major and enabled=false

**Comment by carlos-lehmann** [comment]
@rarkins @RahulGautamSingh much appreciate the quick feedback! That matchUpdateTypes was definitely a wrong interpretation. I didn't even check it as I was so sure it was to match to patch level. /facepalm
Have a great day!
